src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java

```diff
@@ -1,41 +1,483 @@
 package mezz.jei.ingredients;
 
-import java.util.Comparator;
-
 import mezz.jei.config.Constants;
 import mezz.jei.gui.ingredients.IIngredientListElement;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemTool;
+import mezz.jei.config.Config;
+import mezz.jei.util.Translator;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.ai.attributes.AttributeModifier;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemArmor;
+import net.minecraft.item.ItemHoe;
+import net.minecraft.item.ItemShears;
+import scala.Int;
+import java.util.*;
+
+import com.google.common.collect.Multimap;
 
 public final class IngredientListElementComparator implements Comparator<IIngredientListElement> {
 	public static final IngredientListElementComparator INSTANCE = new IngredientListElementComparator();
 
+	public static ArrayList<SortEntry> entries = new ArrayList<SortEntry>();
```

Comment:
1. Field declaration types should not be more specific than necessary.
2. We have java 8, use the diamond operator.

so this should be 
```java
public static List<SortEntry> entries = new ArrayList<>();
```